bottlecap/src/metrics/dogstatsd.rs

```diff
@@ -35,63 +56,115 @@ impl DogStatsD {
         DogStatsD {
             cancel_token,
             aggregator,
-            socket,
-            event_bus,
+            buffer_reader: BufferReader::UdpSocketReader(socket),
         }
     }
 
     pub async fn spin(self) {
-        loop {
-            // TODO(astuyve) this should be dynamic
-            let mut buf = [0; 1024]; // todo, do we want to make this dynamic? (not sure)
-            let (amt, src) = self
-                .socket
-                .recv_from(&mut buf)
-                .await
-                .expect("didn't receive data");
-            let buf = &mut buf[..amt];
-            let msg = std::str::from_utf8(buf).expect("couldn't parse as string");
-            debug!(
-                "received message: {} from {}, sending it to the bus",
-                msg, src
-            );
-            let parsed_metric = match Metric::parse(msg) {
-                Ok(parsed_metric) => {
-                    debug!("parsed metric: {:?}", parsed_metric);
-                    parsed_metric
-                }
+        let mut spin_cancelled = false;
+        while !spin_cancelled {
+            self.consume_statsd().await;
+            spin_cancelled = self.cancel_token.is_cancelled();
+        }
+    }
+
+    async fn consume_statsd(&self) {
+        let (buf, src) = self
+            .buffer_reader
+            .read()
+            .await
+            .expect("didn't receive data");
+        let msgs = std::str::from_utf8(&buf).expect("couldn't parse as string");
+        debug!(
+            "received message: {} from {}, sending it to the bus",
+            msgs, src
```

Comment:
We are not sending anything to a bus anymore tho, maybe update/remove this debug log?

Reply:
I will correct the message. It's still useful for debugging for the source addresser